cmd/minishift/cmd/start_preflight.go

```diff
@@ -22,35 +22,52 @@ import (
 	"os"
 	"os/exec"
 	"regexp"
-	"runtime"
 	"strconv"
 	"strings"
 
-	"github.com/minishift/minishift/pkg/util/filehelper"
-
 	"github.com/docker/machine/libmachine/drivers"
 	configCmd "github.com/minishift/minishift/cmd/minishift/cmd/config"
+	"github.com/minishift/minishift/pkg/minikube/constants"
 	validations "github.com/minishift/minishift/pkg/minishift/config"
 	"github.com/minishift/minishift/pkg/minishift/shell/powershell"
 	minishiftUtil "github.com/minishift/minishift/pkg/minishift/util"
+	"github.com/minishift/minishift/pkg/util/github"
 
 	"github.com/minishift/minishift/pkg/util/os/atexit"
 	"github.com/spf13/viper"
 
 	cmdUtils "github.com/minishift/minishift/cmd/minishift/cmd/util"
-	"github.com/minishift/minishift/pkg/minishift/constants"
+	openshiftVersion "github.com/minishift/minishift/pkg/minishift/openshift/version"
 	stringUtils "github.com/minishift/minishift/pkg/util/strings"
 )
 
 const (
 	StorageDisk = "/mnt/sda1"
 )
 
-// preflightChecksAfterStartingHost is executed before the startHost function.
+// preflightChecksBeforeStartingHost is executed before the startHost function.
 func preflightChecksBeforeStartingHost() {
 	driverErrorMessage := "See the 'Setting Up the Driver Plug-in' topic (https://docs.openshift.org/latest/minishift/getting-started/setting-up-driver-plugin.html) for more information"
 	prerequisiteErrorMessage := "See the 'Installing Prerequisites for Minishift' topic (https://docs.openshift.org/latest/minishift/getting-started/installing.html#install-prerequisites) for more information"
 
+	preflightCheckSucceedsOrFails(
+		configCmd.SkipCheckOpenShiftGithubRelease.Name,
+		checkOcTagOnGithub,
+		fmt.Sprintf("Checking if requested OpenShift version '%s' is released on Github", viper.GetString(configCmd.OpenshiftVersion.Name)),
+		configCmd.WarnCheckOpenShiftGithubRelease.Name,
+		fmt.Sprintf("%s version is not released on Github", viper.GetString(configCmd.OpenshiftVersion.Name)),
```

Comment:
IMO, this message should be extended to include `invalid` version.
```
$ minishift start --openshift-version v1.4.2
-- Starting profile 'minishift'
-- Checking if requested OpenShift version 'v1.4.2' is released on Github ... FAIL
   v1.4.2 version is not released on Github
```
The message `v1.4.2 version is not released on Github` is weird at this point since `v1.4.2` can't be released.

`v1.4.2 version is not a valid OpenShift version on Github`

/cc @robin-owen

Reply:
"%s is not a valid OpenShift version on GitHub" makes more sense to me. Also, I'm being a little pedantic here, but we should capitalize the site's name as "GitHub". :)

Reply:
+1 for `GitHub`.